src/tests/functional/shared_test_classes/src/single_op/matrix_nms.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/single_op/matrix_nms.hpp"
+#include "common_test_utils/test_enums.hpp"
+
+namespace ov {
+namespace test {
+std::string MatrixNmsLayerTest::getTestCaseName(const testing::TestParamInfo<NmsParams>& obj) {
+    std::vector<InputShape> shapes;
+    InputTypes input_types;
+    op::v8::MatrixNms::SortResultType sort_result_type;
+    element::Type outType;
+    int backgroudClass;
+    op::v8::MatrixNms::DecayFunction decayFunction;
+    TopKParams top_k_params;
+    ThresholdParams threshold_params;
+    bool normalized;
+    std::string targetDevice;
+    std::tie(shapes, input_types, sort_result_type, outType, top_k_params, threshold_params,
+        backgroudClass, normalized, decayFunction, targetDevice) = obj.param;
+
+    ElementType paramsPrec, maxBoxPrec, thrPrec;
+    std::tie(paramsPrec, maxBoxPrec, thrPrec) = input_types;
+
+    int nmsTopK, keepTopK;
+    std::tie(nmsTopK, keepTopK) = top_k_params;
+
+    float score_threshold, gaussian_sigma, post_threshold;
+    std::tie(score_threshold, gaussian_sigma, post_threshold) = threshold_params;
+
+    std::ostringstream result;
+    result << "IS=(";
+    for (const auto& shape : shapes) {
+        result << ov::test::utils::partialShape2str({shape.first}) << "_";
+    }
+    result << ")_TS=(";
+    for (const auto& shape : shapes) {
+        for (const auto& item : shape.second) {
+            result << ov::test::utils::vec2str(item) << "_";
+        }
+    }
+
+    using ov::test::utils::operator<<;
+    result << ")_paramsPrec=" << paramsPrec << "_maxBoxPrec=" << maxBoxPrec << "_thrPrec=" << thrPrec << "_";
+    result << "sortResultType=" << sort_result_type << "_normalized=" << normalized << "_";
+    result << "outType=" << outType << "_nmsTopK=" << nmsTopK << "_keepTopK=" << keepTopK << "_";
+    result << "backgroudClass=" << backgroudClass << "_decayFunction=" << decayFunction << "_";
+    result << "score_threshold=" << score_threshold << "_gaussian_sigma=" << gaussian_sigma << "_";
+    result << "post_threshold=" << post_threshold <<"_TargetDevice=" << targetDevice;
+    return result.str();
+}
+
+void MatrixNmsLayerTest::SetUp() {
+    std::vector<InputShape> shapes;
+    InputTypes input_types;
+    TopKParams top_k_params;
+    ThresholdParams threshold_params;
+    ov::op::v8::MatrixNms::Attributes attrs;
+
+    std::tie(shapes, input_types, attrs.sort_result_type, attrs.output_type, top_k_params, threshold_params,
+        attrs.background_class, attrs.normalized, attrs.decay_function, targetDevice) = this->GetParam();
+
+    std::tie(attrs.nms_top_k, attrs.keep_top_k) = top_k_params;
+    std::tie(attrs.score_threshold, attrs.gaussian_sigma, attrs.post_threshold) = threshold_params;
+
+    init_input_shapes(shapes);
+
+    ElementType paramsPrec, maxBoxPrec, thrPrec;
```

Comment:
`maxBoxPrec, thrPrec` not used in the test, should be removed

Reply:
done